smolyay/samples.py

```diff
@@ -0,0 +1,362 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points to be used as the sampling points
+    in a tensor product grid or sparse grid. These unidimensional points are
+    associated with certain polynomial families, and the combination of
+    unique points and polynomial families are the basis for different types
+    of quadrature rules that are used in numerical integration and in
+    approximation.
+    The natural domain of the points specified by :param:``domain`` is
+    the domain in which all the possible points in the set lies within.
+    ``points`` is the list of points that the UnidimensionalPointSet
+    is expected to store.
+    ``domain`` is the domain of the points generated by the point set.
```

Comment:
I think most of this documentation is not current, so I would remove it for now.

Just FYI: attributes and properties are usually also documented where they are defined, rather than at the high level of the class, so that is why I'm suggesting to remove those here.
```suggestion
```